### PR TITLE
Factor out testing code from charts

### PR DESCRIPTION
**Pull request by unguiculus**
Factors out chart testing code into a separate library which is provided as Docker image. This also unifies originally separate and somewhat redundant scripts for CircleCI and CI on our Kubernetes cluster.

### CODE REVIEW

**Comment by unguiculus** [comment]
https://github.com/kubernetes/charts/pull/5231 updates CI in the charts repo.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    py-crcmod \
+    py-pip
+
+RUN STABLE=$(curl -Ls https://storage.googleapis.com/kubernetes-release/release/stable.txt) && \
+    curl -LO https://storage.googleapis.com/kubernetes-release/release/$STABLE/bin/linux/amd64/kubectl && \
+    chmod +x kubectl && \
+    mv kubectl /usr/local/bin/
+
+# Install Helm
+ARG HELM_VERSION=2.8.2
```
Comment (viglesiasce) [inline Dockerfile]: We should do Helm last as that is the most likely to change.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    curl \
+    git \
+    jq \
+    libc6-compat \
+    openssh-client \
+    python \
+    py-crcmod \
+    py-pip
+
+RUN STABLE=$(curl -Ls https://storage.googleapis.com/kubernetes-release/release/stable.txt) && \
```
Comment (viglesiasce) [inline Dockerfile]: Lets hard code the version so we know whats in the image based on the commit/tag

File: README.md [hunk 3]
```diff
@@ -1,3 +1,142 @@
 # Chart Testing
 
-WIP
+Bash library for linting and testing Helm charts. Comes prepackaged as Docker image for easy use.
+
+[chartlib.sh](lib/chartlib.sh) is a Bash library with useful function for linting and testing charts. It is well documented and should be easily usable. The script is meant to be sourced and can be configured via environment variables.
+
+As a convenience, [chartlib.sh](chart_test.sh) is provided. It supports linting and testing charts that have changed against a target branch.
```
Comment (viglesiasce) [inline README.md]: I think this should be [chart_test.sh]

**Comment by viglesiasce** [comment]
Thanks @unguiculus!!

**Comment by unguiculus** [comment]
cc @lachie83 @prydonius @scottrigby

**Comment by prydonius** [comment]
+1 from me, I think @mattfarina also wanted to take a quick look

**Comment by rendhalver** [comment]
This is awesome!

**Pull request merged**